src/utils/telemetry.ts

```diff
@@ -151,22 +157,21 @@ export module Telemetry {
             return userType;
         }
 
-        private static getRegistryValue(key: string, value: string, hive: string): Q.Promise<string> {
-            let deferred: Q.Deferred<string> = Q.defer<string>();
-            let regKey = new winreg({
-                hive: hive,
-                key: key,
-            });
-            regKey.get(value, function (err: any, itemValue: winreg.RegistryItem) {
-                if (err) {
-                    // Fail gracefully by returning null if there was an error.
-                    deferred.resolve(null);
-                } else {
-                    deferred.resolve(itemValue.value);
-                }
+        private static getRegistryValue(key: string, value: string, hive: string): Promise<string> {
+            return new Promise((resolve, reject) => {
+                let regKey = new winreg({
+                    hive: hive,
+                    key: key,
+                });

```

Comment:
Maybe we may use a property shorthand here?
```suggestion
                let regKey = new winreg({ hive, key });

```